src/helium_follower_service.erl

```diff
@@ -54,8 +55,50 @@ subnetwork_last_reward_height(Ctx, Req) ->
 
 -spec active_gateways(follower_pb:follower_gateway_stream_req_v1_pb(), grpcbox_stream:t()) ->
         {ok, grpcbox_stream:t()} | grpcbox_stream:grpc_error_response().
+active_gateways(#follower_gateway_stream_req_v1_pb{} = _Msg, StreamState) ->
+    #{chain := Chain} =  grpcbox_stream:stream_handler_state(StreamState),
+    case Chain of
+        undefined ->
+            lager:debug("chain not ready, returning error response for msg ~p", [_Msg]),
+            {grpc_error, {grpcbox_stream:code_to_status(14), <<"temporarily unavailable">>}};
+        _ ->
+            grpcbox_stream:update_headers([{<<"gateway_stream">>, <<"start">>}], StreamState),
+
+            Ledger = blockchain:ledger(Chain),
+            {ok, Height} = blockchain_ledger_v1:current_height(Ledger),
+
+            InteractiveBlock = case ?get_var(?hip17_interactivity_blocks, Ledger) of
+                                   {ok, V} -> V;
+                                   {error, not_found} -> 0
+                               end,
+            lager:info("LOADING ACTIVE GATEWAYS", []),
+            NumGateways =
+                blockchain_ledger_v1:cf_fold(
+                    active_gateways,
+                    fun({Addr, BinGw}, Acc) ->
+                        Gw = blockchain_ledger_gateway_v2:deserialize(BinGw),
+                        Loc = blockchain_ledger_gateway_v2:location(Gw),
+                        LastChallenge = blockchain_ledger_gateway_v2:last_poc_challenge(Gw),
+                        case Loc /= undefined andalso (LastChallenge /= undefined andalso (Height - LastChallenge) =< InteractiveBlock) of
+                            true ->
+                                Msg = #follower_gateway_resp_v1_pb{
+                                          height = Height,
+                                          address = Addr,
+                                          location = h3:to_string(Loc),
+                                          owner = blockchain_ledger_gateway_v2:owner_address(Gw)
+                                      },
+                                grpcbox_stream:send(false, Msg, StreamState),
+                                Acc + 1;
+                            _ -> Acc
+                        end
+                    end, 0, Ledger),
+
+            grpcbox_stream:update_trailers([{<<"gateway_stream">>, <<"end">>},
+                                         {<<"num_gateways">>, integer_to_binary(NumGateways)}], StreamState),
+            {stop, StreamState}
```

Comment:
so the idea is that you want to iterate over the gateways and send 1 grpc response per gateway ?....for 1m gateways that is a lotta msgs

Would this not be better handled via a unary RPC ?  You could build a list of the gateways, serialize it and then gzip ?

Reply:
i'm trying to avoid either the server or the client having to load the entire list into memory all at once since that _is_ such a big list of gateways. i don't see why we couldn't build up a sub-list of gateways in "chunks" to send it as fewer messages but streaming each gateway individually i can build the resulting hashmap on the rust client side one gateway at a time more or less as they're read out of the ledger from the serverside

Reply:
i see your point re the entire list as a single chunk...not sure if thats worse or better that handling 1m msgs tho

the chunks approach seems like a possible alternative solution but still feels very error prone giving the number of individual msgs going out over the wire.....

Maybe an alternative solution would be to export all the gateways to CSV, read that line by line and add to the hexmap ? Update the CSV periodically, have something write it out every few hours or whatever timeframe required

Reply:
that seems like a hack too.. what's more error prone in the messages over the wire case? How about sending batches of  1000 per response to reduce the number of messages if you're concerned? Actually not sure what the concern is

Reply:
i don't know what's "reasonable" to be concerned about here since "the wire" should theoretically never actually leave `localhost`; once i can get the stream working at all i'd like to see what the response time and result set actually turns out to be. they're filtered through "active gateways" so i believe the rough actual number is closer to 750k

Reply:
k so if its always the localhost thats not as bad...by error prone i mean subject to drop outs, disconnects etc...having to allow for those will make it difficult to ever be sure you have the complete set of gateways.

I think some of the same uncertainty remains over localhost but obviously less prone to connection issues